src/config.py

```diff
@@ -333,21 +354,54 @@ class YDNConfig(BaseConfig):
     is_object_tracker: bool = False
 
     blob_path: str
-    label_map: List[str]
+    labels: List[str]
     confidence_threshold: float = 0.25
     iou_threshold: float = 0.5
     anchors: List[float]
     anchor_masks: Mapping[str, List[int]]
     coordinate_size: int = 4
 
+    # Used in OAK to check what service this config is for
+    service_name: str
+    service_id: str
+
+    @classmethod
+    def from_kwargs(cls, **kwargs):
+        self = cls(dict(), kwargs["service_name"])
+        self.input_source = kwargs["input_source"]
+        self.width = kwargs["width"]
+        self.height = kwargs["height"]
+        self.num_threads = kwargs.get("num_threads", self.num_threads)
+        self.num_nce_per_thread = kwargs.get(
+            "num_nce_per_thread", self.num_nce_per_thread
+        )
+        self.is_object_tracker = kwargs.get("is_object_tracker", self.is_object_tracker)
+
+        self.blob_path = kwargs["blob_path"]
+        self.labels = kwargs["labels"]
+        self.confidence_threshold = kwargs.get(
+            "confidence_threshold", self.confidence_threshold
+        )
+        self.iou_threshold = kwargs.get("iou_threshold", self.iou_threshold)
+        self.anchors = kwargs.get("anchors", [])
+        self.anchor_masks = kwargs.get("anchor_masks", dict())
+        for l in self.anchor_masks.values():
+            for i, num in enumerate(l):
+                l[i] = int(num)
+        self.coordinate_size = kwargs.get("coordinate_size", self.coordinate_size)
+
+        self.service_name = kwargs["service_name"]
+        self.service_id = kwargs["service_id"]
+        return self
+
     @classmethod
     def validate(self, attribute_map: Mapping[str, Value]) -> List[str]:
         # Validate "input_source"
         validate_attr_type("input_source", "string_value", attribute_map, True)
         input_source = attribute_map.get("input_source", default=None).string_value
-        if input_source not in ["cam_a", "cam_b", "cam_c", "color", "depth"]:
+        if input_source not in ["cam_a", "cam_b", "cam_c", "color"]:
```

Comment:
[q] If the YDN config validate fails? Does the camera still work ok?

Reply:
Yup should be decoupled.